src/codec_aom.c

```diff
@@ -783,6 +780,9 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
                 return AVIF_RESULT_UNKNOWN_ERROR;
             }
         }
+
+        // Sync our copy with updated config in AOM encoder.
+        *cfg = *codec->internal->encoder.config.enc;
```

Comment:
Can this be avoided? We should treat `codec->internal->encoder` as an opaque structure. See these comments in aom/aom_codec.h:

```
/*!\brief Codec context structure
 *
 * All codecs \ref MUST support this context structure fully. In general,
 * this data should be considered private to the codec algorithm, and
 * not be manipulated or examined by the calling application. Applications
 * may reference the 'name' member to get a printable description of the
 * algorithm.
 */
typedef struct aom_codec_ctx {
  const char *name;             /**< Printable interface name */
  aom_codec_iface_t *iface;     /**< Interface pointers */
  aom_codec_err_t err;          /**< Last returned error */
  const char *err_detail;       /**< Detailed info, if available */
  aom_codec_flags_t init_flags; /**< Flags passed at init time */
  union {
    /**< Decoder Configuration Pointer */
    const struct aom_codec_dec_cfg *dec;
    /**< Encoder Configuration Pointer */
    const struct aom_codec_enc_cfg *enc;
    const void *raw;
  } config;               /**< Configuration pointer aliasing union */
  aom_codec_priv_t *priv; /**< Algorithm private storage */
} aom_codec_ctx_t;
```

Reply:
`aom_codec_set_option()` can change some options in `cfg`. I didn't find API functions that can get current `cfg` from codec. If we don't update our copy here, these options will only apply to next frame.